# <font color='Black'>Exploratory analysis of data analysts</font>

In this notebook we will perform Exploratory Data Analysis (EDA) that  will focus on overall Data Analyst job title, job description, salary estimate. 
    

<div align ="center">
<img src="https://editor.analyticsvidhya.com/uploads/71710meme-2.jpeg" width="200" height="150" align="center"/>
    </div>

lets import the necessary libraries and load the data


In [2]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import plotly.figure_factory as ff
import plotly.express as px
import plotly.graph_objs as go
from plotly.subplots import make_subplots
import plotly.offline as po


In [3]:
data = pd.read_csv(r'D:/suhas_github/suhas/Artificial-Intelligence/Exploratory Data Analysis/data analyst/data/DataAnalyst.csv')
data.head(10)

Unnamed: 0                                          Job Title  \
0           0  Data Analyst, Center on Immigration and Justic...   
1           1                               Quality Data Analyst   
2           2  Senior Data Analyst, Insights & Analytics Team...   
3           3                                       Data Analyst   
4           4                             Reporting Data Analyst   
5           5                                       Data Analyst   
6           6                       Business/Data Analyst (FP&A)   
7           7                               Data Science Analyst   
8           8                                       Data Analyst   
9           9                      Data Analyst, Merchant Health   

              Salary Estimate  \
0  $37K-$66K (Glassdoor est.)   
1  $37K-$66K (Glassdoor est.)   
2  $37K-$66K (Glassdoor est.)   
3  $37K-$66K (Glassdoor est.)   
4  $37K-$66K (Glassdoor est.)   
5  $37K-$66K (Glassdoor est.)   
6  $37K-$66K (Glassdoor est.)   
7  $37K-$66K (Glassdoor est.)   
8  $37K-$66K (Glassdoor est.)   
9  $37K-$66K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Are you eager to roll up your sleeves and harn...     3.2   
1  Overview\n\nProvides analytical and technical ...     3.8   
2  We’re looking for a Senior Data Analyst who ha...     3.4   
3  Requisition NumberRR-0001939\nRemote:Yes\nWe c...     4.1   
4  ABOUT FANDUEL GROUP\n\nFanDuel Group is a worl...     3.9   
5  About Cubist\nCubist Systematic Strategies is ...     3.9   
6  Two Sigma is a different kind of investment ma...     4.4   
7  Data Science Analyst\n\nJob Details\nLevel\nEx...     3.7   
8  The Data Analyst is an integral member of the ...     4.0   
9  About Us\n\nRiskified is the AI platform power...     4.4   

                              Company Name      Location  \
0           Vera Institute of Justice\n3.2  New York, NY   
1  Visiting Nurse Service of New York\n3.8  New York, NY   
2                         Squarespace\n3.4  New York, NY   
3                            Celerity\n4.1  New York, NY   
4                             FanDuel\n3.9  New York, NY   
5                             Point72\n3.9  New York, NY   
6                           Two Sigma\n4.4  New York, NY   
7             GNY Insurance Companies\n3.7  New York, NY   
8                                DMGT\n4.0  New York, NY   
9                           Riskified\n4.4  New York, NY   

             Headquarters                     Size  Founded  \
0            New York, NY     201 to 500 employees     1961   
1            New York, NY         10000+ employees     1893   
2            New York, NY   1001 to 5000 employees     2003   
3              McLean, VA     201 to 500 employees     2002   
4            New York, NY    501 to 1000 employees     2009   
5            Stamford, CT   1001 to 5000 employees     2014   
6            New York, NY   1001 to 5000 employees     2001   
7            New York, NY     201 to 500 employees     1914   
8  London, United Kingdom  5001 to 10000 employees     1896   
9            New York, NY    501 to 1000 employees     2013   

                Type of ownership                               Industry  \
0          Nonprofit Organization                      Social Assistance   
1          Nonprofit Organization       Health Care Services & Hospitals   
2               Company - Private                               Internet   
3  Subsidiary or Business Segment                            IT Services   
4               Company - Private                    Sports & Recreation   
5               Company - Private  Investment Banking & Asset Management   
6               Company - Private  Investment Banking & Asset Management   
7               Company - Private                     Insurance Carriers   
8                Company - Public       Venture Capital & Private Equity   
9               Company - Private                 Research & Developme

Lets get the basic information about the dataset

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2253 entries, 0 to 2252
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         2253 non-null   int64  
 1   Job Title          2253 non-null   object 
 2   Salary Estimate    2253 non-null   object 
 3   Job Description    2253 non-null   object 
 4   Rating             2253 non-null   float64
 5   Company Name       2252 non-null   object 
 6   Location           2253 non-null   object 
 7   Headquarters       2253 non-null   object 
 8   Size               2253 non-null   object 
 9   Founded            2253 non-null   int64  
 10  Type of ownership  2253 non-null   object 
 11  Industry           2253 non-null   object 
 12  Sector             2253 non-null   object 
 13  Revenue            2253 non-null   object 
 14  Competitors        2253 non-null   object 
 15  Easy Apply         2253 non-null   object 
dtypes: float64(1), int64(2),

We can see that there is a total of 15 columns and 2253 rows. "company name" seems to be having missing values. The columns "Salary Estimate","Size" and "Revenue" are of object datatype and not integer.

Hence we can preprocces the data according to our needs before we explore the data or use it for analysis. The main steps to prepare a dataset are:
 - look for missing values
 - look for the area of interest( based on the aim of the project)


<div align ="center">
<img src="https://media.makeameme.org/created/lets-take-a-92541440de.jpg" width="200" height="150" align="center"/>
    </div>

In [5]:
data.isnull().sum()

Unnamed: 0           0
Job Title            0
Salary Estimate      0
Job Description      0
Rating               0
Company Name         1
Location             0
Headquarters         0
Size                 0
Founded              0
Type of ownership    0
Industry             0
Sector               0
Revenue              0
Competitors          0
Easy Apply           0
dtype: int64

As discussed above "company name" has one missing value. but if we see the top 5 rows itself we can find certain columns filled with data as "-1". 
In real-world datasets, most of the time the missing data is not recorded like software recognizable version of missing values, hence we have to check the data and find them.

For now lets change "-1" as missing value(Nan).
We have two datatype with -1 values, hence we will have to convert the string and numeric verion of -1 to Nan.

In [6]:
data.replace('-1',np.nan,inplace =True)
data.replace(-1,np.nan,inplace =True)

Now lets check the number of missing values in the dataset

In [8]:
data.isnull().sum()

Unnamed: 0              0
Job Title               0
Salary Estimate         1
Job Description         0
Rating                272
Company Name            1
Location                0
Headquarters          172
Size                  163
Founded               660
Type of ownership     163
Industry              353
Sector                353
Revenue               163
Competitors          1732
Easy Apply           2173
dtype: int64

<div align ="center">
<img src="https://i.kym-cdn.com/entries/icons/facebook/000/023/180/notsurprisedkirk.jpg" width="200" height="150" align="center"/>
    </div>

And That is the surprising reality of real world data. Some columns have a huge number of missing values. Hence the importance of preprocessing your data. Based on our aim, we can select which data is required, which data needs to be processed and which can be eliminated from the dataset

In this project we will focus on the following columns:
 - Job Title
 - Salary Estimate
 - Job Description
 - Rating
 - Company name
 - Industry
 - Sector

let us dive into the data,explore these columns and modify them accordingly

In [10]:
# check the different jobs 
data['Job Title'].value_counts()

Data Analyst                                 405
Senior Data Analyst                           90
Junior Data Analyst                           30
Business Data Analyst                         28
Sr. Data Analyst                              21
                                            ... 
Junior Data Analyst - SQL Programmer           1
Data Analyst - Aurora, IL                      1
Business Analyst/Data Consultant               1
Sr. Analyst, Product Content Data Quality      1
Data Development Analyst~                      1
Name: Job Title, Length: 1272, dtype: int64

In [11]:
data['Job Title'].value_counts()[:20]

Data Analyst               405
Senior Data Analyst         90
Junior Data Analyst         30
Business Data Analyst       28
Sr. Data Analyst            21
Data Analyst II             17
Data Quality Analyst        17
Data Analyst Junior         17
Data Governance Analyst     16
Lead Data Analyst           15
Data Reporting Analyst      13
Financial Data Analyst      12
Data Analyst III            11
Data Analyst I              11
Marketing Data Analyst       9
Sr Data Analyst              9
Data Warehouse Analyst       8
Data Management Analyst      8
Data Science Analyst         7
Technical Data Analyst       7
Name: Job Title, dtype: int64

We can modify the above data to classify senior and junior properly


In [14]:
data["Job Title"] = data["Job Title"].str.replace('Sr. Data Analyst','Senior Data Analyst')
data["Job Title"] = data["Job Title"].str.replace('Sr Data Analyst','Senior Data Analyst')
data["Job Title"] = data["Job Title"].str.replace('Data Analyst Senior','Senior Data Analyst')
data["Job Title"] = data["Job Title"].str.replace('Jr. Data Analyst','Junior Data Analyst')
data["Job Title"] = data["Job Title"].str.replace('Jr Data Analyst','Junior Data Analyst')
data["Job Title"] = data["Job Title"].str.replace('Data Analyst Junior','Junior Data Analyst')
data["Job Title"].value_counts()

Data Analyst                                          405
Senior Data Analyst                                   121
Junior Data Analyst                                    50
Business Data Analyst                                  28
Data Quality Analyst                                   17
                                                     ... 
Security Data Analyst -CTJ with Security Clearance      1
Data Analyst - Contract Renewals                        1
Data Control Business Analyst                           1
Junior Data Analyst - SQL Programmer                    1
Data Development Analyst~                               1
Name: Job Title, Length: 1266, dtype: int64

In [15]:
data["Job Title"].isnull().sum() # no missing values found in the j ob title

0

Lets move on to the Salary column

In [16]:
data["Salary Estimate"].isnull().sum()

1

As we can see that there is one single missing value in salary.Its a very important column for analysis and hence we cannot ignore the missing value. Even if you have no domain experience we can fill in the missing value by checking at similar jobs from the same company. 

Still filling the value is a complex task as it contains text, numberic and $ value. so lets see how we can proceed

In [17]:
#explore the missing value row
data[data['Salary Estimate'].isnull()]

Unnamed: 0                                  Job Title Salary Estimate  \
2149        2149  Configuration/Data Management Analyst III             NaN   

                                        Job Description  Rating  \
2149  Job Description\nPosition Title: Configuration...     4.4   

         Company Name  Location  Headquarters                 Size  Founded  \
2149  Protingent\n4.4  Kent, WA  Bellevue, WA  51 to 200 employees   2001.0   

      Type of ownership                Industry             Sector  \
2149  Company - Private  Staffing & Outsourcing  Business Services   

                       Revenue Competitors Easy Apply  
2149  $25 to $50 million (USD)         NaN        NaN

In [21]:
""" The company name is Protingent\n4.4 and job title is Data management analyst |||. lets filter these values"""
data[data['Company Name']=='Protingent\n4.4']

Unnamed: 0                                          Job Title  \
2123        2123  Engines Configuration and Data Management Analyst   
2149        2149          Configuration/Data Management Analyst III   

                  Salary Estimate  \
2123  $55K-$101K (Glassdoor est.)   
2149                          NaN   

                                        Job Description  Rating  \
2123  Job Description\nPosition Title: Engines Confi...     4.4   
2149  Job Description\nPosition Title: Configuration...     4.4   

         Company Name  Location  Headquarters                 Size  Founded  \
2123  Protingent\n4.4  Kent, WA  Bellevue, WA  51 to 200 employees   2001.0   
2149  Protingent\n4.4  Kent, WA  Bellevue, WA  51 to 200 employees   2001.0   

      Type of ownership                Industry             Sector  \
2123  Company - Private  Staffing & Outsourcing  Business Services   
2149  Company - Private  Staffing & Outsourcing  Business Services   

                       Revenue Competitors Easy Apply  
2123  $25 to $50 million (USD)         NaN        NaN  
2149  $25 to $50 million (USD)         NaN        NaN

In [22]:
data[data['Job Title']=='Configuration/Data Management Analyst III']

Unnamed: 0                                  Job Title Salary Estimate  \
2149        2149  Configuration/Data Management Analyst III             NaN   

                                        Job Description  Rating  \
2149  Job Description\nPosition Title: Configuration...     4.4   

         Company Name  Location  Headquarters                 Size  Founded  \
2149  Protingent\n4.4  Kent, WA  Bellevue, WA  51 to 200 employees   2001.0   

      Type of ownership                Industry             Sector  \
2149  Company - Private  Staffing & Outsourcing  Business Services   

                       Revenue Competitors Easy Apply  
2149  $25 to $50 million (USD)         NaN        NaN

We can observe that there only one  requirement for that job and there are 2 job requirements from the same company. if the job description are similar then we can set the same pricing

In [27]:
print(data['Job Description'][2123])
data['Job Description'][2149]

Job Description
Position Title: Engines Configuration and Data Management Analyst

Position Description: Protingent Staffing has an exciting contract opportunity with our client in Kent, WA.

Job Qualifications:
AA/AS or BA/BS in a related technical or liberal arts discipline
3+ years of direct product configuration management experience (Product structure, parts, drawings, documents and requirements management)
2+ years of experience working in a highly regulated environment (aerospace, nuclear, government, medical device, etc.)
Experience working with software related to configuration management of complex product
Working knowledge of configuration management principles and practices
Familiar with product lifecycle management (PLM) and/or data management tools
Possess high attention to detail and strong organizational/coordination skills
Proficient in Microsoft Office applications (Word, Excel, PowerPoint).
Experience in aerospace industry
Proficient in technical writing of process a

'Job Description\nPosition Title: Configuration/Data Management Analyst III\n\nPosition Description: Protingent has an opportunity for a Configuration Analyst in Kent, WA.\n\nJob Qualifications:\nAA/AS or BA/BS in a related technical or liberal arts discipline\n5-7 years of direct product configuration management experience (Product structure, parts, drawings and requirements management experience highly desired).\n3+ years of experience working in a highly regulated environment (security systems, medical device, aerospace, etc.).\nDeep experience with world-class toolsets for configuration management in complex products\nProven ability to communicate effectively both in writing and verbally with staff at all levels of the organization and external stakeholders, including regulatory agencies, customers and suppliers.\nExcellent technical writing and presentation skills.\nProficient with configuration management principles and practices\nProficient with product lifecycle management (PLM

Even though thereare similarities i guess we can drop the row with the missing value. 

Furthermore as it is a salary estimate column we can further get more data from it. We can calculate Max salary, Min salary and average salary.

In [28]:
data.drop(2149, inplace=True)

In [29]:
data['max salary'] = data['Salary Estimate'].str[6:10].str.replace('K','').str.lstrip('$').str.strip().astype("float")
data['max salary']

0        66.0
1        66.0
2        66.0
3        66.0
4        66.0
        ...  
2248    104.0
2249    104.0
2250    104.0
2251    104.0
2252    104.0
Name: max salary, Length: 2252, dtype: float64

In [30]:
data["min salary"] = data['Salary Estimate'].str.lstrip('$').str[:3].str.replace('K','').str.strip().astype('float')
data['min salary']

0       37.0
1       37.0
2       37.0
3       37.0
4       37.0
        ... 
2248    78.0
2249    78.0
2250    78.0
2251    78.0
2252    78.0
Name: min salary, Length: 2252, dtype: float64

In [31]:
data["avg salary"] = (data["min salary"]+ data['max salary'])/2
data["avg salary"]

0       51.5
1       51.5
2       51.5
3       51.5
4       51.5
        ... 
2248    91.0
2249    91.0
2250    91.0
2251    91.0
2252    91.0
Name: avg salary, Length: 2252, dtype: float64

In [33]:
data[['Salary Estimate','min salary','max salary','avg salary']].sample(10)

Salary Estimate  min salary  max salary  avg salary
1005  $46K-$102K (Glassdoor est.)        46.0       102.0        74.0
1539  $69K-$127K (Glassdoor est.)        69.0       127.0        98.0
1947  $93K-$159K (Glassdoor est.)        93.0       159.0       126.0
1872   $53K-$99K (Glassdoor est.)        53.0        99.0        76.0
1629  $63K-$116K (Glassdoor est.)        63.0       116.0        89.5
874    $68K-$87K (Glassdoor est.)        68.0        87.0        77.5
2096   $35K-$67K (Glassdoor est.)        35.0        67.0        51.0
1047   $43K-$94K (Glassdoor est.)        43.0        94.0        68.5
1377   $41K-$86K (Glassdoor est.)        41.0        86.0        63.5
506    $30K-$54K (Glassdoor est.)        30.0        54.0        42.0

Moving on to job description, we can extract valuable information from this column. Here we can extract different language requirements for the job.

In [35]:
data["Job Description"].isnull().sum()

0

In [36]:
data['python job'] = data["Job Description"].str.contains('python', na =False, case =False)
data["python job"].value_counts()

False    1615
True      637
Name: python job, dtype: int64

In [38]:
data['java job'] = data["Job Description"].str.contains('java', na =False, case =False)
data['java job'].value_counts()

False    2048
True      204
Name: java job, dtype: int64

In [39]:
data['sql job'] = data["Job Description"].str.contains('sql', na =False, case =False)
data['sql job'].value_counts()

True     1388
False     864
Name: sql job, dtype: int64

In [40]:
data['tableau job'] = data["Job Description"].str.contains('tabeau', na =False, case =False)
data['tableau job'].value_counts()

False    2252
Name: tableau job, dtype: int64

In [41]:
data['excel job'] = data["Job Description"].str.contains('excel', na =False, case =False)
data['excel job'].value_counts()

True     1353
False     899
Name: excel job, dtype: int64

In [42]:
data['HTML job'] = data["Job Description"].str.contains('HTML', na =False, case =False)
data['HTML job'].value_counts()

False    2221
True       31
Name: HTML job, dtype: int64

In [45]:
# moving on to rating
print(data['Rating'].isnull().sum())
# we will keep the rows which has missing value on Rating, we ant to use salary and job title info on these rows.
data["Company Name"].sample(10)

272


1632                                   Apple\n4.1
235                                       2U\n3.5
322                         Quad Consultancy\n4.3
1667                           911 Datamaster Inc
208     Royal Communications Consultants Inc\n3.0
1898                                  PubNub\n4.1
184                22nd Century Technologies\n3.7
2008                                    Vida\n3.6
751                        First Stop Health\n4.6
1790           Buckeye Community Health Plan\n3.4
Name: Company Name, dtype: object

We can see that the company seems to have ratings attached to it. once we verify this we can clean the company name column

In [46]:
data[["Company Name","Rating"]].sample(10)

Company Name  Rating
1317          Delmarva Foundation\n3.1     3.1
2140                 Ostendo Solutions     NaN
791                     Invenergy\n3.8     3.8
1291             Web Shop Manager\n4.2     4.2
1259                          Cue\n4.2     4.2
950         Kelsey Seybold Clinic\n3.6     3.6
589   STAND 8 Technology Services\n3.9     3.9
2141                     W3Global\n4.5     4.5
1506      Lorven Technologies Inc\n4.0     4.0
2040                 Square Trade\n3.5     3.5

In [47]:
data['Company Name'] = data['Company Name'].str.split('\n').str[0]
data['Company Name'].sample(10)

1067                  ECRI
1456          FirstAmerica
577                 Acorns
1554    Armada Group, Inc.
948        Infomerica, Inc
1760           CyberCoders
1795           Wells Fargo
418         ACT Consulting
1334             Precision
10      NYU Langone Health
Name: Company Name, dtype: object

In [48]:
#we had seen that a single data is missing in company name
data[data["Company Name"].isnull()]
# We can keep this row, for job title, salary and job description

Unnamed: 0     Job Title             Salary Estimate  \
1860        1860  Data Analyst  $53K-$99K (Glassdoor est.)   

                                        Job Description  Rating Company Name  \
1860  Kindred at Home, part of the Kindred at Home f...     NaN          NaN   

             Location Headquarters Size  Founded  ... Easy Apply max salary  \
1860  Mooresville, NC          NaN  NaN      NaN  ...        NaN       99.0   

     min salary avg salary python job java job  sql job  tableau job  \
1860       53.0       76.0      False    False     True        False   

      excel job  HTML job  
1860       True     False  

[1 rows x 25 columns]

In [50]:
data.columns

Index(['Unnamed: 0', 'Job Title', 'Salary Estimate', 'Job Description',
       'Rating', 'Company Name', 'Location', 'Headquarters', 'Size', 'Founded',
       'Type of ownership', 'Industry', 'Sector', 'Revenue', 'Competitors',
       'Easy Apply', 'max salary', 'min salary', 'avg salary', 'python job',
       'java job', 'sql job', 'tableau job', 'excel job', 'HTML job'],
      dtype='object')

In [53]:
analysis_data =data[[ 'Job Title','Rating', 'Company Name', 'Industry', 'Sector', 'Revenue', 'max salary', 'min salary',
                     'avg salary', 'python job','java job', 'sql job', 'tableau job', 'excel job', 'HTML job']]
analysis_data.head()

Job Title  Rating  \
0  Data Analyst, Center on Immigration and Justic...     3.2   
1                               Quality Data Analyst     3.8   
2  Senior Data Analyst, Insights & Analytics Team...     3.4   
3                                       Data Analyst     4.1   
4                             Reporting Data Analyst     3.9   

                         Company Name                          Industry  \
0           Vera Institute of Justice                 Social Assistance   
1  Visiting Nurse Service of New York  Health Care Services & Hospitals   
2                         Squarespace                          Internet   
3                            Celerity                       IT Services   
4                             FanDuel               Sports & Recreation   

                             Sector                     Revenue  max salary  \
0                        Non-Profit  $100 to $500 million (USD)        66.0   
1                       Health Care      $2 to $5 billion (USD)        66.0   
2            Information Technology    Unknown / Non-Applicable        66.0   
3            Information Technology   $50 to $100 million (USD)        66.0   
4  Arts, Entertainment & Recreation  $100 to $500 million (USD)        66.0   

   min salary  avg salary  python job  java job  sql job  tableau job  \
0        37.0        51.5        True     False     True        False   
1        37.0        51.5       False     False     True        False   
2        37.0        51.5        True     False     True        False   
3        37.0        51.5       False     False     True        False   
4        37.0        51.5        True     False     True        False   

   excel job  HTML job  
0      False     False  
1       True     False  
2       True     False  
3      False     False  
4       True     False

In [55]:
analysis_data.isnull().sum()

Job Title         0
Rating          272
Company Name      1
Industry        353
Sector          353
Revenue         163
max salary        0
min salary        0
avg salary        0
python job        0
java job          0
sql job           0
tableau job       0
excel job         0
HTML job          0
dtype: int64

In [56]:
analysis_data.describe()

Rating   max salary   min salary   avg salary
count  1980.000000  2252.000000  2252.000000  2252.000000
mean      3.731566    89.979130    54.266874    72.123002
std       0.670333    29.315605    19.575375    23.600734
min       1.000000    38.000000    24.000000    33.500000
25%       3.300000    70.000000    41.000000    58.000000
50%       3.700000    87.000000    50.000000    69.000000
75%       4.100000   104.000000    64.000000    80.500000
max       5.000000   190.000000   113.000000   150.000000

With that we are done with processing and modifying the data according to our requirements. We have taken only the columns which can be used for analysis in a new dataframe.Let's look at the some of the information we can get from above table:
 - Average and also median value for the ratings of the companies is close to  4 
 - Average minimum salary is around 54K and median value for minimum salary is around 50K. 
 - Average maximum salary is almost 90K and median value for maximum salary is 87K.
 - Expect outliers from salary distribution 
 - Expect right skewed distribution of the minimum and maximum salary
 - Average salary is around 72K

<div align ="center">
<img src="https://memegenerator.net/img/instances/44911040.jpg" width="200" height="150" align="center"/>
    </div>

##### Lets start visualizing the data to get more insights on the data

In [58]:
salary_min = px.histogram(analysis_data,
                             x= 'min salary',
                             title='Minimum Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = analysis_data[['Job Title', 'Company Name']])
salary_max = px.histogram(analysis_data,
                             x= 'max salary',
                             title='Maximum Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = analysis_data[['Job Title', 'Company Name']])
salary_avg = px.histogram(analysis_data,
                             x= 'avg salary',
                             title='Average Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = analysis_data[['Job Title', 'Company Name']])

In [59]:
salary_min.show()
salary_max.show()
salary_avg.show()

### Salary visualization

- As expected we have a skewed distribution and a few outliers. 
- The majority of the minimum salary count lies between 40k-65k and maximum salary count lies between 65K-105k.
- On average The salaries lie in between 55k-85k

Now lets find out the distribution of Data analysts only.

In [61]:
data_analyst = analysis_data[analysis_data["Job Title"]=='Data Analyst']
data_analyst.head()

Job Title  Rating        Company Name  \
3   Data Analyst     4.1            Celerity   
5   Data Analyst     3.9             Point72   
8   Data Analyst     4.0                DMGT   
10  Data Analyst     4.0  NYU Langone Health   
11  Data Analyst     NaN            BulbHead   

                                 Industry                  Sector  \
3                             IT Services  Information Technology   
5   Investment Banking & Asset Management                 Finance   
8        Venture Capital & Private Equity                 Finance   
10       Health Care Services & Hospitals             Health Care   
11                                    NaN                     NaN   

                      Revenue  max salary  min salary  avg salary  python job  \
3   $50 to $100 million (USD)        66.0        37.0        51.5       False   
5    Unknown / Non-Applicable        66.0        37.0        51.5        True   
8      $1 to $2 billion (USD)        66.0        37.0        51.5        True   
10    $5 to $10 billion (USD)        66.0        37.0        51.5       False   
11   Unknown / Non-Applicable        66.0        37.0        51.5        True   

    java job  sql job  tableau job  excel job  HTML job  
3      False     True        False      False     False  
5       True     True        False       True     False  
8      False     True        False       True     False  
10     False    False        False       True     False  
11     False     True        False       True     False

In [62]:
data_analyst.describe()

Rating  max salary  min salary  avg salary
count  349.000000  405.000000  405.000000  405.000000
mean     3.850143   90.874074   54.296296   72.585185
std      0.679529   35.892185   21.089181   28.007461
min      1.000000   38.000000   24.000000   33.500000
25%      3.400000   67.000000   38.000000   53.500000
50%      3.900000   85.000000   51.000000   68.000000
75%      4.300000  103.000000   60.000000   79.500000
max      5.000000  190.000000  110.000000  150.000000

In [63]:
salary_min = px.histogram(data_analyst,
                             x= 'min salary',
                             title='Minimum Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = data_analyst[['Job Title', 'Company Name']])
salary_max = px.histogram(data_analyst,
                             x= 'max salary',
                             title='Maximum Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = data_analyst[['Job Title', 'Company Name']])
salary_avg = px.histogram(data_analyst,
                             x= 'avg salary',
                             title='Average Salary distribution of Data Analyst Jobs',
                             marginal="box",
                             hover_data = data_analyst[['Job Title', 'Company Name']])
salary_min.show()
salary_max.show()
salary_avg.show()

### Salary visualization
- Average and also median value for the ratings of the companies very close to 4
- As expected we have a skewed distribution and a few outliers. 
- The average minimum salary is 54k and average maximum salary  is 85k.
- On average The salaries are 75k


In [ ]:
# visualizing the top 20 jobs with number of available spots
jobs_figure = go.Figure()
jobs_figure.add_trace(go.Scatter(
  x = analysis_data['Job Title'].value_counts()[:20].index,
  y = analysis_data['Job Title'].value_counts()[:20].values,
  name = "Number of Job openings ",
  mode ='markers+text+lines',
  marker_color ="black",
  marker_size = 10,
  text = analysis_data['Job Title'].value_counts()[:20].values,
  textposition ='top center',
  line=dict(color='red',dash='dash'),
))
jobs_figure.update_layout(
  title ="<b>number of job openings by job titles</b>",
  xaxis_title ="<b>job titles</b>",
  yaxis_title ="<b>Job Openings</b>",
  template ="seaborn",
  font =dict(
    size =12,
    color ='Black',
    family ='Oswald, sans-serif',
  ),
  xaxis=dict(showgrid=True),
  yaxis=dict(showgrid=True),
  yaxis2=dict(showgrid=True,overlaying='y',side='right',title='<b>Number of Job Openings</b>'),
  legend=dict(yanchor="top",
              y=1.3,
              xanchor="left",
              x=0.78)
  )

In [ ]:
# visualizing the top 20 Industry with number of available spots
industry_fig = go.Figure()
industry_fig.add_trace(go.Scatter(
  x = analysis_data['Industry'].value_counts()[:20].index,
  y = analysis_data['Industry'].value_counts()[:20].values,
  name = "Number of Job openings ",
  mode ='markers+text+lines',
  marker_color ="black",
  marker_size = 10,
  text = analysis_data['Industry'].value_counts()[:20].values,
  textposition ='top center',
  line=dict(color='blue',dash='dash'),
))
industry_fig.update_layout(
  title ="<b>number of job openings by Industry</b>",
  xaxis_title ="<b>Industry</b>",
  yaxis_title ="<b>Job Openings</b>",
  template ="seaborn",
  font =dict(
    size =12,
    color ='Black',
    family ='Oswald, sans-serif',
  ),
  xaxis=dict(showgrid=True),
  yaxis=dict(showgrid=True),
  yaxis2=dict(showgrid=True,overlaying='y',side='right',title='<b>Number of Job Openings</b>'),
  legend=dict(yanchor="top",
              y=1.3,
              xanchor="left",
              x=0.78)
  )

In [ ]:
# visualizing the top 20 sectors with number of available spots
Sector_fig = go.Figure()
Sector_fig.add_trace(go.Scatter(
  x = analysis_data['Sector'].value_counts()[:20].index,
  y = analysis_data['Sector'].value_counts()[:20].values,
  name = "Number of Job openings ",
  mode ='markers+text+lines',
  marker_color ="black",
  marker_size = 10,
  text = analysis_data['Sector'].value_counts()[:20].values,
  textposition ='top center',
  line=dict(color='blue',dash='dash'),
))
Sector_fig.update_layout(
  title ="<b>number of job openings by Sector</b>",
  xaxis_title ="<b>Sector</b>",
  yaxis_title ="<b>Job Openings</b>",
  template ="seaborn",
  font =dict(
    size =12,
    color ='Black',
    family ='Oswald, sans-serif',
  ),
  xaxis=dict(showgrid=True),
  yaxis=dict(showgrid=True),
  yaxis2=dict(showgrid=True,overlaying='y',side='right',title='<b>Number of Job Openings</b>'),
  legend=dict(yanchor="top",
              y=1.3,
              xanchor="left",
              x=0.78)
  )


In [92]:
jobs_figure.show()
industry_fig.show()
Sector_fig.show()

From the above figures we can see that:
- Data Analyst,Senior Data Analyst,Junior Data Analyst,Business Data Analyst are the most used **titles in the job advertisements.**
- IT Services,Staffing & Outsourcing,Health Care Services & Hospitals,Consulting,Computer Hardware & Software are the **most data analyst job opening advertised industries.**
- Information Technology,Business Services,Finance,Health Care are the most **data analyst job opening advertised sectors.**

now lets have a look at the different language requirement for the jobs


In [93]:
df = analysis_data[['Job Title','python job','sql job','java job','HTML job','tableau job','excel job']].copy()

In [99]:
language = df.groupby('Job Title')[['python job','sql job','java job','HTML job','tableau job','excel job']].sum().sort_values(by='python job',ascending = False).head(10)

In [101]:
dfL = pd.DataFrame(language)
dfL = dfL.reset_index()
dfL['job openings'] = analysis_data['Job Title'].value_counts()[:10].values
columnsTitles = ['Job Title', 'job openings','python job','sql job','java job','HTML job','tableau job','excel job']

dfL = dfL.reindex(columns=columnsTitles)

dfL

Job Title  job openings  \
0                                       Data Analyst           405   
1                                Senior Data Analyst           121   
2                              Business Data Analyst            50   
3               NY Healthcare Data/Reporting Analyst            28   
4                             Marketing Data Analyst            17   
5            Data Analyst III (Healthcare Analytics)            17   
6                                  Lead Data Analyst            16   
7               TX Healthcare Data/Reporting Analyst            15   
8  Data Analyst, Data & Analytics (Advanced Analy...            13   
9                  Healthcare Data/Reporting Analyst            12   

   python job  sql job  java job  HTML job  tableau job  excel job  
0         125      301        54         5            0        239  
1          68      104         9         1            0         67  
2          11       21         1         0            0         17  
3           5        5         0         0            0          5  
4           5        9         0         0            0          6  
5           5        5         0         0            0          5  
6           5       13         1         0            0          7  
7           4        4         0         0            0          4  
8           4        4         0         0            0          0  
9           4        4         0         0            0          4

In [102]:
fig = px.bar(dfL, x='Job Title', y=['python job','sql job','java job','HTML job','tableau job','excel job'], title="Languages")
fig.show()